src/util/std_types.cpp

```diff
@@ -283,3 +283,19 @@ bool is_constant_or_has_constant_components(
 
   return false;
 }
+
+vector_typet::vector_typet(const typet &_subtype, const constant_exprt &_size)
+  : type_with_subtypet(ID_vector, _subtype)
+{
+  size() = _size;
+}
```

Comment:
What's the rationale to move these to the .cpp file?

Reply:
`constant_exprt` isn't complete -- there's a cyclic dependency between std_types.h and std_expr.h.